constellations/gemini-admin/src/home/home.js

```diff
@@ -2,107 +2,126 @@
-                            eruditi copiosae contentiones cu. Sed ea quando
-                            mediocrem, soleat dolorum no ius. Eu electram
-                            iracundia mnesarchum his, eum ut hinc latine. Est
-                            simul definiebas ut, debitis invenire eu usu. Eu
-                            eros appetere mel, nullam delenit tincidunt duo at.
-                            Eos option appetere torquatos in, cu debitis
-                            singulis principes usu.
-                        </p>
-                    </div>
-                </div>
+                <div className="showSection">{sectionComponent}</div>
+            </div>
+        );
+    }
+}
 
-                <div className="section">
-                    <h3 className="section-header">New Registrations</h3>
-                    <div className="container-registration">
-                        <p>
-                            Lorem ipsum dolor sit amet, vim ei tota dicant
-                            interpretaris, sea consulatu scripserit ei, pri eu
-                            accumsan contentiones. Usu ei iriure deleniti. Cum
-                            ut periculis laboramus referrentur. Vis ea laoreet
-                            imperdiet deterruisset, probo pertinax iudicabit qui
-                            ea. Duo liber quodsi contentiones ne, vel ei movet
-                            conceptam, id his quod iriure feugiat. Qui ex vide
-                            labitur volumus. His cu fugit adolescens
-                            voluptatibus, et per torquatos interesset. Iusto
-                            deleniti invenire id cum. Te saepe alterum
-                            appellantur pro, in est errem dicant suscipit,
-                            oblique argumentum sed et. Quo cu partem inermis,
-                            vix et minimum vivendum, ut illud delectus eos. Mei
-                            te alia justo, per amet nemore quodsi cu, mel
-                            eruditi copiosae contentiones cu. Sed ea quando
-                            mediocrem, soleat dolorum no ius. Eu electram
-                            iracundia mnesarchum his, eum ut hinc latine. Est
-                            simul definiebas ut, debitis invenire eu usu. Eu
-                            eros appetere mel, nullam delenit tincidunt duo at.
-                            Eos option appetere torquatos in, cu debitis
-                            singulis principes usu.
-                        </p>
-                    </div>
-                </div>
+class TabButton extends React.Component {
+    state = {
+        isZero: false,
+    };
+
+    componentDidMount = () => {
+        if (this.props.buttonNum == 0) {
+            this.setState({
+                isZero: true,
+            });
+        }
+        console.log("isZero state is " + this.state.isZero);
+    };
```

Comment:
i think it's not working because `componentDidMount` only runs once, and because `buttonNum` comes from an async function this all runs before the actual number is passed in. I'm not sure if using state is really necessary, having isZero as a local variable in render might be easier

tip: `setState` is an async function which is why the `console.log` is always showing false. you can things run after setState by passing in a function as the second argument, e.g.
```
this.setState({
    isZero: true,
}, () => {
    console.log("isZero state is " + this.state.isZero)
})
```

Reply:
+1 I would get rid of this.state.isZero and instead, you can just always check if this.props.buttonNum is equal to 0. By introducing a new state variable that is similar to this props variable, you introduce weird states you could potentially be in (i.e. `buttonNum` is 0, but there's a small window of time that `isZero` is not 0).

So below, just have:
```
const numNotif = this.props.buttonNum;
...
<div className={"notif" + (numNotif == 0 ? " zero" : "")}>
                {numNotif}
</div>
```